test/sanity/validate-modules/main.py

```diff
@@ -375,14 +375,24 @@ def _check_for_sys_exit(self):
                 msg='sys.exit() call found. Should be exit_json/fail_json'
             )
 
-    def _check_for_gpl3_header(self):
-        if ('GNU General Public License' not in self.text and
-                'version 3' not in self.text):
+    def _check_gpl3_header(self):
+        header = '\n'.join(self.text.split('\n')[:20])
+        if ('GNU General Public License' not in header and
+                'version 3' not in header):
             self.reporter.error(
                 path=self.object_path,
                 code=105,
-                msg='GPLv3 license header not found'
+                msg='GPLv3 license header not found in the first 20 lines of the module'
             )
+        if self._is_new_module():
```

Comment:
I'd restructure the test to be:

```python
if self._is_new_module():
   # new module test here
else:
    # old module test here
```

That way you can't end up with two license errors reported for a new module.